dhctl/pkg/kubernetes/actions/converge/node.go

```diff
@@ -76,6 +76,56 @@ func GetCloudConfig(kubeCl *client.KubernetesClient, nodeGroupName string) (stri
 	return cloudData, err
 }
 
+func GetNodeTemplate(kubeCl *client.KubernetesClient, nodeGroupName string) (map[string]interface{}, error) {
+	var res map[string]interface{}
+	err := retry.NewSilentLoop(fmt.Sprintf("Get NodeGroup %q", nodeGroupName), 45, 15*time.Second).Run(func() error {
+		ng, err := kubeCl.Dynamic().
+			Resource(nodeGroupResource).
+			Get(context.TODO(), nodeGroupName, metav1.GetOptions{})
+
+		if err != nil {
+			return err
+		}
+
+		nt, _, err := unstructured.NestedMap(ng.Object, "spec", "nodeTemplate")
+		if err != nil {
+			return err
+		}
+
+		res = nt
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func UpdateNodeTemplate(kubeCl *client.KubernetesClient, nodeGroupName string, template map[string]interface{}) error {
+	return retry.NewLoop(fmt.Sprintf("Update node template in NodeGroup %q", nodeGroupName), 45, 15*time.Second).Run(func() error {
+		doc, err := kubeCl.Dynamic().
```

Comment:
You get node group manifest twice. The first is to get the node template section, and the second is for the update operation.

I believe there is a logical bug.

1. dhctl gets node template from the cluster
2. dhctl shows a diff to the user and asks for confirmation
3. Someone changes the node template in the cluster
4. User confirms node template updating and wipes the previously added date

If you get a node group from the cluster only once, you will not face this bug due to resource version inconsistency, which is the correct behavior from my point of view.

Reply:
Good point! Thank you. I fixed it.